packages/samples/headless-react/src/components/search-parameter-manager/search-parameter-manager.ts

```diff
@@ -1,19 +1,30 @@
 import {
+  AnalyticsActions,
   buildSearchParameterManager,
   buildSearchParameterSerializer,
   Engine,
+  SearchActions,
 } from '@coveo/headless';
 
 const {serialize, deserialize} = buildSearchParameterSerializer();
 
-export function bindSearchParametersToURI(engine: Engine) {
+export function bindSearchParametersToURI(
+  engine: Engine,
+  options: {executeInitialSearch: boolean}
+) {
   const hash = window.location.hash.slice(1);
   const parameters = deserialize(decodeURIComponent(hash));
 
   const searchParameterManager = buildSearchParameterManager(engine, {
     initialState: {parameters},
   });
 
+  if (options.executeInitialSearch) {
+    engine.dispatch(
+      SearchActions.executeSearch(AnalyticsActions.logInterfaceLoad())
+    );
+  }
```

Comment:
I made `executeInitialSearch` a required option because I wanted it to be obvious in App.tsx and in `search-parameter-manager.ts` where the initial search should be executed.

With this sample, we also see that the initial search must be executed after building the search parameter manager, not before.

Reply:
Why should the search-parameter-manager execute the search? Keep in mind that this sample is being used as the reference for the SearchParameterManager controller. I feel adding this code here makes the sample a little misleading?

Could we place the code in the App.tsx lifecycle?

```
// App.tsx

componentDidMount() {
    bindSearchParametersToURI(engine)
    executeFirstSearch()
}

```

Reply:
We could, but I also find intuitive the concept that we read the parameters from the URL and use those to start a search (which is essentially what the engine ends up doing).

If those are moved to `App.tsx`, I believe a comment should be added that mentions the first search **must** be executed after the SearchParametersManager was built.

Reply:
Yeah, a comment is a good idea, and we should also probably add this note in the controller's documentation.

The initialize sequence is:

1. All components must register on the engine.
2. Only then should one restore the url parameters.
3. Execute search.

One of the main reasons to use `componentDidMount` is to ensure that all the App's children have registered on the engine. If the restoration happens earlier, it will not succeed because e.g the facet id in the url will not be in the store.